pkgs/by-name/op/opencl-caps-viewer/package.nix

```
@@ -0,0 +1,73 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  ocl-icd,
+  opencl-headers,
+  libsForQt5,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opencl-caps-viewer";
+  version = "1.20";
+
+  src = fetchFromGitHub {
+    owner = "SaschaWillems";
+    repo = "OpenCLCapsViewer";
+    tag = finalAttrs.version;
+    hash = "sha256-P7G8FvVXzDAfN3d4pGXC+c9x4bY08/cJNYQ6lvjyVCQ=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    libsForQt5.qmake
+    libsForQt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    ocl-icd
+    opencl-headers
+    libsForQt5.qtx11extras
+    libsForQt5.qtbase
+  ];
+
+  patchPhase = ''
+    # Fix installation paths
+    substituteInPlace OpenCLCapsViewer.pro \
+      --replace-fail "target.path = /usr/bin" "target.path = /bin/" \
+      --replace-fail "desktop.path = /usr/share/applications" "desktop.path = /share/applications" \
+      --replace-fail "icon.path = /usr/share/icons/hicolor/256x256/apps/" "icon.path = /share/icons/hicolor/256x256/apps/" \
+      --replace-fail 'else: unix:!android: target.path = /opt/$${TARGET}/bin' ""
+  '';
+
+  qmakeFlags = [
+    "OpenCLCapsViewer.pro"
+    "CONFIG+=x11"
+  ];
+
+  installFlags = [ "INSTALL_ROOT=$(out)" ];
```

```suggestion
  installFlags = [ "INSTALL_ROOT=${placeholder "out"}" ];
```

Using bash syntax in other strings is risky because sometimes `lib.escapeShellArgs` is processing them